### ISSUE TITLE
Discrepancy standardize_parameters()

### ISSUE THREAD

**Issue by strengejacke** [easystats/parameters#36]
I think these three methods should return the same results, however, `standardize_parameters()` differs (for *factors*).

``` r
library(lm.beta)
library(sjstats)
library(parameters)
#> 
#> Attaching package: 'parameters'
#> The following object is masked from 'package:sjstats':
#> 
#>     p_value

data(iris)
m <- lm(Sepal.Length ~ Species + Petal.Width, data = iris)

lm.beta(m)
#> 
#> Call:
#> lm(formula = Sepal.Length ~ Species + Petal.Width, data = iris)
#> 
#> Standardized Coefficients::
#>       (Intercept) Speciesversicolor  Speciesvirginica       Petal.Width 
#>        0.00000000       -0.03441674       -0.02860860        0.84401156

std_beta(m)
#>                term std.estimate std.error   conf.low conf.high
#> 1 Speciesversicolor  -0.03441674 0.1316090 -0.2923655 0.2235321
#> 2  Speciesvirginica  -0.02860860 0.2046162 -0.4296489 0.3724317
#> 3       Petal.Width   0.84401156 0.1784474  0.4942611 1.1937620

standardize_parameters(m, method = "full", robust = FALSE)
#>           Parameter Std_Estimate
#> 1       (Intercept)           NA
#> 2 Speciesversicolor  -0.07276516
#> 3  Speciesvirginica  -0.06048538
#> 4       Petal.Width   0.84401156
```

<sup>Created on 2019-05-01 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by DominiqueMakowski** [comment]
I think this is related to [this](https://github.com/easystats/parameters/issues/31#issuecomment-484945492) comparison. 
Long story short, the present standardization treats factor levels accordingly, and only standardizes the coefs by the SD of the outcome. On the contrary, if I recall correctly, other packages also standardize by the SD of the factor (whatever that means).

**Comment by strengejacke** [comment]
AH, I remember... We should add some more details to the _Details_ section in the docs explaining the differences. Do you want to make a first draft, or should I start?

**Comment by DominiqueMakowski** [comment]
Please feel free to start (I am still a bit swamped 🏊), but I'll complete ASAP!

**Comment by DominiqueMakowski** [comment]
I started looking at the docs for standardization and the associated vignette. 

I wonder, what are the applications for 2sd method? 
Ans also, why return `Std_Estimate` instead of the previous behaviour, which was `Std_*` + the name of the estimate? Which seemed more in line with the output of parameters and the possibility for Bayesian or bootstrapped models to have multiple estimates (median, MAP mean etc)

**Comment by strengejacke** [comment]
I'm not sure, I think that `Std_Estimate` was your initial name given? Or maybe there was no name, and I harmonized it with functions like `model_performance()`.  There was no special intention. But since it's a column that applies to all parameters, `Std_*`+name in most cases makes no sense?

**Comment by strengejacke** [comment]
> I wonder, what are the applications for 2sd method?

http://www.stat.columbia.edu/~gelman/research/published/standardizing7.pdf

**Comment by DominiqueMakowski** [comment]
> To think
of this another way, consider a regression with some binary predictors (for example, a male/female indicator) left intact, and some continuous predictors (for example, height) scaled by dividing by one standard deviation. The coefficients for the binary predictors correspond to a comparison of
x =0 to x =1, or two standard deviations.
For these reasons, we recommend the general practice of scaling numeric inputs by dividing by two standard deviations, which allows the coefficients to be interpreted in the same way as with binary inputs. (We leave binary inputs unscaled because their coefficients can already be
interpreted directly.)

"a comparison of
x =0 to x =1, or two standard deviations" I am not sure I get that part

**Comment by strengejacke** [comment]
See following example:

``` r
for (i in seq(.2, .8, .1)) {
  x <- sample(x = c(0, 1), size = 1000, replace = TRUE, prob = c(i, 1 - i))
  cat(sprintf("Probalility of success: %.0f%%\n", 100 * (1 - i)))
  print(sd(x))
}
#> Probalility of success: 80%
#> [1] 0.4067978
#> Probalility of success: 70%
#> [1] 0.4535247
#> Probalility of success: 60%
#> [1] 0.4873074
#> Probalility of success: 50%
#> [1] 0.500025
#> Probalility of success: 40%
#> [1] 0.4886612
#> Probalility of success: 30%
#> [1] 0.4623124
#> Probalility of success: 20%
#> [1] 0.3876781
```

No matter which distribution you have, the standard deviation of a binary variable is roughly ~0.5. If you standardize the binary variable, the standardized estimate of this variable corresponds to the change of 1 SD in your original variable. For binary variables, this estimate would mean a difference between `x = 0` and `x = 0.5`  (=1SD). Since the binary variable has the value 0 and 1, this makes not much sense.

If you standardize by _two_ SD, you would compare `x = 0` to `x = 2 * 0.5` (=2SD), which is you compare `x=0` to `x=1` (which makes sense for binary variables).

So the idea is, since dividing by 2 SD, you don't need to standardize binary variables, because you compare 0 to 1 in either way. But to compare the estimate ("effect size") of binary variables to continuous variables, these need to be divided by 2 SD as well.

To summarize: you just need to divide continuous variables by 2 SD, and this standardized estimate is directly comparable to the (unstandardized!) estimate of binary variables. What do we gain? The inference for continuous variables is similar: A unit change in _two_ (instead of _one_) SD means a unit-change in the outcome. But estimates of binary variables are directly interpretable.

**Comment by DominiqueMakowski** [comment]
Thanks a lot for the clarification!
